File: 2843_count_symmetric_integers.cpp
class Solution {
public:
    int countSymmetricIntegers(int l, int h) {
        int z=0;
        for(int x=l;x<=h;++x){
            string y=to_string(x);
            int n=y.size(),a=0,b=0;
            if(n%2)continue;
            for(int i=0;i<n/2;++i)a+=y[i]-48,b+=y[i+n/2]-48;
            z+=a==b;
        }
        return z;
    }
};

File: 2843_count_symmetric_integers.java
class Solution {
    public int countSymmetricIntegers(int x, int y) {
        int g = 0;
        for (int i = x; i <= y; i++) {
            String s = String.valueOf(i);
            int l = s.length(), a = 0;
            if (l % 2 == 1) continue;
            for (int j = 0; j < l / 2; j++) a += s.charAt(j) - s.charAt(j + l / 2);
            if (a == 0) g++;
        }
        return g;
    }
}
